include/Beman/Execution26/detail/fwd_env.hpp

```diff
@@ -0,0 +1,43 @@
+// include/Beman/Execution26/detail/fwd_env.hpp                       -*-C++-*-
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef INCLUDED_BEMAN_EXECUTION26_DETAIL_FWD_ENV
+#define INCLUDED_BEMAN_EXECUTION26_DETAIL_FWD_ENV
+
+#include <Beman/Execution26/detail/common.hpp>
+#include <Beman/Execution26/detail/forwarding_query.hpp>
+#include <type_traits>
+#include <utility>
+
+// ----------------------------------------------------------------------------
+
+namespace Beman::Execution26::Detail
+{
+    template <typename Env>
+    class fwd_env
+    {
+    private:
+        Env env;
+    
+    public:
+        fwd_env(Env&& env): env(::std::forward<Env>(env)) {}
+
+        template <typename Query, typename... Args>
+            requires (not ::Beman::Execution26::forwarding_query(::std::remove_cvref_t<Query>()))
+        constexpr auto query(Query&& q, Args&&... args) const
+            = BEMAN_EXECUTION26_DELETE("the used query is not forwardable");
```

Comment:
delete with a message -- implemented with compatibility?

Reply:
Well, [`delete` with message](http://wg21.link/p2573)  isn't implemented in any of the compilers I'm actually using (it seems it is coming with [gcc](http://gcc.gnu.org/)-15 and [clang](http://llvm.org)-19). I should probably build a current version and see if it actually works.